[Aaron Loh] iP

aaronloh852 opened the pull request:
# This is a Heading

* _This is bullet point no.1_
* __This is bullet point no. 2__

1. ~~This is a numbered list point no.1~~
2. This is a numbered list point no. 2
3. This is a numbered list point no. 3

```
if (isFencedBlockCode){
  return true;
}
```
- [x] This is a complete item
- [x] This is an incomplete item

This is an emoji :smiley_cat:

> This is a quote.
> - Person who said the quote

[GitHub](http://github.com)

This is an example of an inline code 
`main` over here.

Commit:
Add Gradle support

aaronloh852 committed:
Level 1, Added Greet, Echo and Exit

aaronloh852 committed:
Level 1, Greet, Echo, Exit

aaronloh852 committed:
Level 2, Add, List

aaronloh852 committed:
Level 3, Mark as Done, new Task class

aaronloh852 committed:
Level 4, ToDos, Events, Deadlines

aaronloh852 committed:
A-TextUiTesting, added input and expected text files

aaronloh852 committed:
Level 5, Error Handling

aaronloh852 committed:
Level 6, added Delete functionality

aaronloh852 committed:
Shift processing of strings and task array into functions
Added javadocs

aaronloh852 committed:
Duke class: Add javadocs, and save system for list of tasks

aaronloh852 committed:
Add ability for deadline and event tasks to recognise dates and times
Add a new keyword "check" to list all deadlines and events occuring on a specified date

aaronloh852 committed:
Merge branch 'branch-Level-7'

aaronloh852 committed:
Resolve merge conflict between branch-Level-7 and branch-Level-8

aaronloh852 committed:
Extract functionality into different classes (Ui, Storage, Parser, TaskList) to be in line with OOP framework

aaronloh852 committed:
Extract code functionality to handle in different classes (Ui, Storage, Parser, TaskList) so as to be in line with OOP framework

aaronloh852 committed:
Redistribute classes into respective packages

aaronloh852 committed:
Add JUnit tests for Parser and Command classes
Update file path for duke class

aaronloh852 committed:
Add JavaDoc comments to the code

aaronloh852 committed:
Update code to follow coding standards

aaronloh852 committed:
Add the find functionality

aaronloh852 committed:
Add the find functionality

aaronloh852 committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

# Conflicts:
#	src/main/java/duke/command/CommandHelp.java

aaronloh852 committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

aaronloh852 committed:
Add JavaDoc comments

aaronloh852 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

aaronloh852 committed:
Add check to see if time argument in Event Task comprises a start and end time

aaronloh852 committed:
Add checkstyle config files and edited src files accordingly to adhere to coding standards

aaronloh852 committed:
Add GUI with the use of JavaFX
Remove CommandBye.java due to redundancy

joshenx commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+public class Storage {
+    private Scanner s;
+    private final String filePath;
+    private File savefile;
```
Perhaps a better name is ```saveFile```?

joshenx commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,97 @@
+     */
+    public void updateSavefile(TaskList tl) {
+        try {
+            FileWriter myWriter = new FileWriter("saves/saves.txt");
```
Is there any reason for naming it ```myWriter``` instead of ```writer```?

joshenx commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+     *
+     * @return A boolean indicating if the arguments are in a valid format.
+     */
+    public boolean isArgumentValid() {
```
Perhaps a better method name is ```isValidArgument```? Same may apply to the other methods related to the ```Command``` class!

joshenx commented on src/main/java/duke/command/CommandFind.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    public void execute(TaskList tl, Storage st, Ui ui) {
+        if (isArgumentValid()) {
+            tl.printAllTasksWith(arguments.get(0));
```
Perhaps a better name is ```printTasksContainingPhrase``` or ```printTasksContainingWord```?

joshenx commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+
+    @Override
+    public String getReadableString() {
+        String status = this.isDone ? "1" : "0";
```
Could rename to ```binaryStatus```?

joshenx reviewed (approved):
LGTM, I only have a few nitpicks regarding the naming of some variables. Otherwise, code looks very clean and well presented! Good job :)

hanif-kamal commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    /**
+     * This method runs the program.
```
Maybe this could just be changed to:
```suggestion
     * Runs the program.
```

hanif-kamal commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+     * If it does, return its index number in the TASK_TYPES array.
+     * If it does not, return -1.
+     *
+     * @param s  String to be checked.
```
Is there an additional whitespace here?
```suggestion
     * @param s String to be checked.
```

hanif-kamal commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+    /**
+     * Takes the user input and returns a valid command if it has been typed.
+     *
+     * @param userInput  String representing input typed in by user.
```
Is there an additional whitespace here?
```suggestion
     * @param userInput String representing input typed in by user.
```

hanif-kamal commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,141 @@
+
+    /**
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
```
Maybe there should be an empty line between the description and parameter section? Also, it appears the parameter isDobie is not described in this JavaDoc. 
```suggestion
     * Creates a label with the specified text and adds it to the dialog container.
     *
     * @param text String containing text to add
     * @param isDobie <description here>
```

hanif-kamal commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,141 @@
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
+     * @param text String containing text to add
+     * @return a label with the specified text that has word wrap enabled.
```
Perhaps you could capitalise the first letter of the sentence after the @return tag like so: 
```suggestion
     * @return A label with the specified text that has word wrap enabled.
```

hanif-kamal reviewed (commented):
Overall it's very well done! There are just a few minor tweaks to consider.

aaronloh852 committed:
Add several assert statements in the main class files

The assert statements are added to help with testing

aaronloh852 committed:
Merge pull request #1 from e0543860/branch-A-Assertions

Add several assert statements in the main class files

aaronloh852 committed:
Improve code quality and made minor edits

aaronloh852 committed:
Add extension D-Contacts

Added a feature to maintain a contact list with adding and deleting functions.

aaronloh852 committed:
Add extension D-Contacts

Add feature to maintain a contacts list with ability to add and delete contacts

aaronloh852 committed:
Merge pull request #2 from e0543860/branch-D-Contacts

Branch D Contacts

aaronloh852 committed:
Add background images and border to dialog boxes
Add javadoc comments to the DialogBox and ContactsList classes

aaronloh852 committed:
Update README.md and Ui.png for iP showcase
Add more commands to CommandHelp class

aaronloh852 committed:
Corrected wrong command format in README.md

aaronloh852 committed:
Set theme jekyll-theme-minimal
